video_inference/examples/demo_camera.py

```diff
@@ -0,0 +1,136 @@
+# ============================================================================== #
+# IMPORTS: DO NOT MODIFY                                                         #
+# ============================================================================== #
+
+import argparse
+import sys
+from typing import Any
+
+from gst.pipeline import GstPipelineGenerator
+from utils.model_info import get_model_input_dims
+from utils.user_input import get_inp_src_info, get_inf_model, validate_inp_dims
+from utils.common import InputType, CAM_DEFAULT_WIDTH, CAM_DEFAULT_HEIGHT
+
+# ============================================================================== #
+
+
+# Your camera's device ID, typically has the format "/dev/videoX". "AUTO" means the script will attempt to find your camera.
+# Note that "AUTO" might find the wrong camera if you have multiple cameras connected to the baord.
+# Execute `v4l2-ctl --list-devices` to get a list of connected camera devices and their device IDs.
+CAMERA = "AUTO"
+
+# Your camera's input width.
+INPUT_WIDTH = CAM_DEFAULT_WIDTH   # 640
+
+# Your camera's input height.
+INPUT_HEIGHT = CAM_DEFAULT_HEIGHT # 480
+
+# The path to the inference model to use. Must be a vaild SyNAP model with a ".synap" file extension.
+MODEL = "/usr/share/synap/models/object_detection/coco/model/yolov8s-640x384/model.synap"
+
+# How many frames to skip between sucessive inferences.
+# Increasing this number may result in better performance but can look worse visually.
+INFERENCE_SKIP = 1
+
+# Maximum number of inference results to display per frame
+MAX_RESULTS = 5
+
+# Confidence threshold, only detections with scores above this will be considered valid
+CONF_THRESHOLD = 0.5
+
+# Whether to launch the demo in fullscreen.
+FULLSCREEN = False
```

Comment:
Setting FULLSCREEN = True by default would make more sense for a better demo experience!

Reply:
Yeah but the issue here is if they run a demo from the a terminal on the board, then fullscreen will cover the terminal, preventing them from closing the demo. Especially for video streams with no definite end like camera or RTSP.